Dockerfile

```diff
@@ -1,74 +1,137 @@
+RUN unzip $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && rm $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && mv projector-server-1.0-SNAPSHOT $PROJECTOR_DIR/ide/projector-server \
+    && mv $PROJECTOR_DIR/ide-projector-launcher.sh $PROJECTOR_DIR/ide/bin \
+    && chmod 644 $PROJECTOR_DIR/ide/projector-server/lib/*
 
-# Copy predefined configs
-COPY --chown=0:0 config/etc /etc/
+# Stage 4. Build the main image with necessary environment for running Projector
+#   Doesn't require to be a desktop environment. Projector runs in headless mode.
+FROM fedora:32
+ARG downloadUrl
+ENV PROJECTOR_USER_NAME projector-user
+ENV PROJECTOR_DIR /projector
+ENV HOME /home/$PROJECTOR_USER_NAME
+ENV PROJECTOR_CONFIG_DIR $HOME/.config
+RUN true \
+    # Any command which returns non-zero exit code will cause this shell script to exit immediately.
+    && set -e \
+    # Activate debugging to show execution details: all commands will be printed before execution.
+    && set -x \
+    # Install necessary packages.
+    && yum install -y \
+    # The non-interactive network downloader.
+    wget \
+    # Git is a fast, scalable, distributed revision control system with an unusually rich command set that provides both high-level operations and full access to internals.
+    git \
+    # Network Security Services (NSS) is a set of libraries designed to support cross-platform development of security-enabled client and server applications. 
+    # Applications built with NSS can support SSL v3, TLS, PKCS #5, PKCS #7, PKCS #11, PKCS #12, S/MIME, X.509 v3 certificates, and other security standards.
+    nss \
+    # Packages required by JetBrains products.
+    libsecret jq e2fsprogs-libs \
+    # Packages needed for AWT.
+    libXext libXrender libXtst libXi freetype \
+    # Collection of shell functions that take advantage of the programmable completion feature of bash 2.04 and later.
+    bash-completion \
+    # Provides command line and full screen utilities for browsing procfs, a "pseudo" file system dynamically generated by the kernel to provide information about the status 
+    # of entries in its process table (such as whether the process is running, stopped, or a "zombie"). Requires by IDEA to disable warnings. 
+    procps \
+    # Provides find utility. Fedora 32 doesn't contain find utility by default.
+    findutils \
+    # Check whether download Url represent IntelliJ IDEA to install required dependencies.
+    && if [ "${downloadUrl#*idea}" != "$downloadUrl" ]; then yum install -y java-11-openjdk-devel; else echo "Not IntelliJ Idea"; fi \
+    # Check whether download Url represent PyCharm to install required dependencies.
+    && if [ "${downloadUrl#*pycharm}" != "$downloadUrl" ]; then yum install -y python2 python3 python3-distutils-extra python3-pip python3-setuptools; else echo "Not Pycharm"; fi
```

Comment:
do we want to support pycharm? or just IDEA?

Reply:
PyCharm Community also free as IntelliJ IDEA Community. So it would be nice to provide the image for it. The resulting image will be the same as for IntelliJ IDEA Community except a few libraries needed for python runtime. The same for IntelliJ IDEA Community, this image has openjdk11.